{
  "issue_number": 5054,
  "title": "test: test cases::test_merge::test_node_request_snapshot_reject_merge failed",
  "body": "```\r\nthread 'raftstore-1-0::test_node_request_snapshot_reject_merge' panicked at 'called `Result::unwrap()` on an `Err` value: \"SendError(..)\"', src/libcore/result.rs:999:5\r\nstack backtrace:\r\n   0: backtrace::backtrace::libunwind::trace\r\n             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/libunwind.rs:88\r\n   1: backtrace::backtrace::trace_unsynchronized\r\n             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/mod.rs:66\r\n   2: std::sys_common::backtrace::_print\r\n             at src/libstd/sys_common/backtrace.rs:47\r\n   3: std::sys_common::backtrace::print\r\n             at src/libstd/sys_common/backtrace.rs:36\r\n   4: std::panicking::default_hook::{{closure}}\r\n             at src/libstd/panicking.rs:198\r\n   5: std::panicking::default_hook\r\n             at src/libstd/panicking.rs:212\r\n   6: std::panicking::rust_panic_with_hook\r\n             at src/libstd/panicking.rs:475\r\n   7: std::panicking::continue_panic_fmt\r\n             at src/libstd/panicking.rs:382\r\n   8: rust_begin_unwind\r\n             at src/libstd/panicking.rs:309\r\n   9: core::panicking::panic_fmt\r\n             at src/libcore/panicking.rs:85\r\n  10: core::result::unwrap_failed\r\n  11: core::result::Result<T,E>::unwrap\r\n  12: <test_raftstore::transport_simulate::RecvSnapshotFilter as test_raftstore::transport_simulate::Filter>::before\r\n  13: test_raftstore::transport_simulate::filter_send\r\n  14: <test_raftstore::transport_simulate::SimulateTransport<C> as tikv::server::transport::RaftStoreRouter>::send_raft_msg\r\n  15: <test_raftstore::node::ChannelTransport as tikv::raftstore::store::transport::Transport>::send\r\n  16: <test_raftstore::transport_simulate::SimulateTransport<C> as tikv::raftstore::store::transport::Transport>::send::{{closure}}\r\n  17: test_raftstore::transport_simulate::filter_send\r\n  18: <test_raftstore::transport_simulate::SimulateTransport<C> as tikv::raftstore::store::transport::Transport>::send\r\n  19: tikv::raftstore::store::peer::Peer::send_raft_message\r\n  20: tikv::raftstore::store::peer::Peer::send\r\n  21: tikv::raftstore::store::peer::Peer::handle_raft_ready_append\r\n  22: tikv::raftstore::store::fsm::peer::PeerFsmDelegate<T,C>::collect_ready\r\n  23: <tikv::raftstore::store::fsm::store::RaftPoller<T,C> as tikv::raftstore::store::fsm::batch::PollHandler<tikv::raftstore::store::fsm::peer::PeerFsm,tikv::raftstore::store::fsm::store::StoreFsm>>::handle_normal\r\n  24: tikv::raftstore::store::fsm::batch::Poller<N,C,Handler>::poll\r\n  25: tikv::raftstore::store::fsm::batch::BatchSystem<N,C>::spawn::{{closure}}\r\nnote: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.\r\nthread 'cases::test_merge::test_node_request_snapshot_reject_merge' panicked at 'called `Result::unwrap()` on an `Err` value: Any', src/libcore/result.rs:999:5\r\nstack backtrace:\r\n   0: backtrace::backtrace::libunwind::trace\r\n             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/libunwind.rs:88\r\n   1: backtrace::backtrace::trace_unsynchronized\r\n             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/mod.rs:66\r\n   2: std::sys_common::backtrace::_print\r\n             at src/libstd/sys_common/backtrace.rs:47\r\n   3: std::sys_common::backtrace::print\r\n             at src/libstd/sys_common/backtrace.rs:36\r\n   4: std::panicking::default_hook::{{closure}}\r\n             at src/libstd/panicking.rs:198\r\n   5: std::panicking::default_hook\r\n             at src/libstd/panicking.rs:212\r\n   6: std::panicking::rust_panic_with_hook\r\n             at src/libstd/panicking.rs:475\r\n   7: std::panicking::continue_panic_fmt\r\n             at src/libstd/panicking.rs:382\r\n   8: rust_begin_unwind\r\n             at src/libstd/panicking.rs:309\r\n   9: core::panicking::panic_fmt\r\n             at src/libcore/panicking.rs:85\r\n  10: core::result::unwrap_failed\r\n  11: core::result::Result<T,E>::unwrap\r\n  12: tikv::raftstore::store::fsm::batch::BatchSystem<N,C>::shutdown\r\n  13: tikv::raftstore::store::fsm::store::RaftBatchSystem::shutdown\r\n  14: tikv::server::node::Node<C>::stop_store\r\n  15: tikv::server::node::Node<C>::stop\r\n  16: <test_raftstore::node::NodeCluster as test_raftstore::cluster::Simulator>::stop_node\r\n  17: test_raftstore::cluster::Cluster<T>::stop_node\r\n             at ./components/test_raftstore/src/cluster.rs:215\r\n  18: test_raftstore::cluster::Cluster<T>::shutdown\r\n             at ./components/test_raftstore/src/cluster.rs:523\r\n  19: <test_raftstore::cluster::Cluster<T> as core::ops::drop::Drop>::drop\r\n             at ./components/test_raftstore/src/cluster.rs:1019\r\n  20: core::ptr::real_drop_in_place\r\n             at /rustc/0e4a56b4b04ea98bb16caada30cb2418dd06e250/src/libcore/ptr/mod.rs:197\r\n  21: failpoints::cases::test_merge::test_node_request_snapshot_reject_merge\r\n             at tests/failpoints/cases/test_merge.rs:470\r\n  22: failpoints::cases::test_merge::test_node_request_snapshot_reject_merge::{{closure}}\r\n             at tests/failpoints/cases/test_merge.rs:434\r\n  23: core::ops::function::FnOnce::call_once\r\n             at /rustc/0e4a56b4b04ea98bb16caada30cb2418dd06e250/src/libcore/ops/function.rs:231\r\n  24: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once\r\n             at /rustc/0e4a56b4b04ea98bb16caada30cb2418dd06e250/src/liballoc/boxed.rs:746\r\n  25: __rust_maybe_catch_panic\r\n             at src/libpanic_unwind/lib.rs:82\r\n  26: std::panicking::try\r\n             at /rustc/0e4a56b4b04ea98bb16caada30cb2418dd06e250/src/libstd/panicking.rs:273\r\n  27: std::panic::catch_unwind\r\n             at /rustc/0e4a56b4b04ea98bb16caada30cb2418dd06e250/src/libstd/panic.rs:388\r\n  28: test::run_test::run_test_inner::{{closure}}\r\n             at src/libtest/lib.rs:1466\r\nnote: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.\r\nthread 'raftstore-2-1::test_node_request_snapshot_reject_merge' panicked at 'called `Result::unwrap()` on an `Err` value: \"PoisonError { inner: .. }\"', src/libcore/result.rs:999:5\r\nstack backtrace:\r\n   0: backtrace::backtrace::libunwind::trace\r\n             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/libunwind.rs:88\r\n   1: backtrace::backtrace::trace_unsynchronized\r\n             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/mod.rs:66\r\n   2: std::sys_common::backtrace::_print\r\n             at src/libstd/sys_common/backtrace.rs:47\r\n   3: std::sys_common::backtrace::print\r\n             at src/libstd/sys_common/backtrace.rs:36\r\n   4: std::panicking::default_hook::{{closure}}\r\n             at src/libstd/panicking.rs:198\r\n   5: std::panicking::default_hook\r\n             at src/libstd/panicking.rs:212\r\n   6: std::panicking::rust_panic_with_hook\r\n             at src/libstd/panicking.rs:475\r\n   7: std::panicking::continue_panic_fmt\r\n             at src/libstd/panicking.rs:382\r\n   8: rust_begin_unwind\r\n             at src/libstd/panicking.rs:309\r\n   9: core::panicking::panic_fmt\r\n             at src/libcore/panicking.rs:85\r\n  10: core::result::unwrap_failed\r\n  11: core::result::Result<T,E>::unwrap\r\n  12: <test_raftstore::node::ChannelTransport as tikv::raftstore::store::transport::Transport>::send\r\n  13: <test_raftstore::transport_simulate::SimulateTransport<C> as tikv::raftstore::store::transport::Transport>::send::{{closure}}\r\n  14: test_raftstore::transport_simulate::filter_send\r\n  15: <test_raftstore::transport_simulate::SimulateTransport<C> as tikv::raftstore::store::transport::Transport>::send\r\n  16: tikv::raftstore::store::peer::Peer::send_raft_message\r\n  17: tikv::raftstore::store::peer::Peer::send\r\n  18: tikv::raftstore::store::peer::Peer::post_raft_ready_append\r\n  19: tikv::raftstore::store::fsm::peer::PeerFsmDelegate<T,C>::post_raft_ready_append\r\n  20: tikv::raftstore::store::fsm::store::RaftPoller<T,C>::handle_raft_ready\r\n  21: <tikv::raftstore::store::fsm::store::RaftPoller<T,C> as tikv::raftstore::store::fsm::batch::PollHandler<tikv::raftstore::store::fsm::peer::PeerFsm,tikv::raftstore::store::fsm::store::StoreFsm>>::end\r\n  22: tikv::raftstore::store::fsm::batch::Poller<N,C,Handler>::poll\r\n  23: tikv::raftstore::store::fsm::batch::BatchSystem<N,C>::spawn::{{closure}}\r\n```",
  "state": "closed",
  "created_at": "2019-07-09T11:36:04Z",
  "updated_at": "2019-07-11T11:31:05Z",
  "closed_at": "2019-07-11T11:31:05Z",
  "labels": [
    "type/bug"
  ],
  "comments_data": [
    {
      "id": 509923354,
      "user": "mapleFU",
      "created_at": "2019-07-10T06:11:50Z",
      "body": "I meet the same problem, and I will list some logs below:\r\n\r\n* https://internal.pingcap.net/idc-jenkins/blue/rest/organizations/jenkins/pipelines/tikv_ghpr_test/runs/6492/nodes/110/steps/658/log/?start=0\r\n* https://internal.pingcap.net/idc-jenkins/blue/rest/organizations/jenkins/pipelines/tikv_ghpr_test/runs/6480/nodes/110/steps/596/log/?start=0\r\n\r\n"
    }
  ]
}